File: cluster-check-go/pods.go
package main

import (
	"context"
	"fmt"
	"strings"

	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
	"k8s.io/client-go/kubernetes"
)

func validatePods(configCheck *Config, clientset *kubernetes.Clientset, sceCheck *bool, azureCheck *bool, omsCheck *bool) bool {
	fmt.Println(Separator)
	fmt.Println("Checking for pods")
	fmt.Println(Separator)
	passed := true

	if *omsCheck {
		if !checkPods(&configCheck.OMS.Pods, clientset) {
			passed = false
		}
	}
	if *sceCheck {
		if !checkPods(&configCheck.SCE.Pods, clientset) {
			passed = false
		}
	}
	if *azureCheck {
		if !checkPods(&configCheck.Azure.Pods, clientset) {
			passed = false
		}
	}
	fmt.Print("\n")
	return passed
}

func checkPods(pods *[]NamespacedEntry, clientset *kubernetes.Clientset) bool {
	var passed = true
	for _, podTest := range *pods {
		status := ""
		color := ""
		pods, err := clientset.CoreV1().Pods(podTest.Namespace).List(context.TODO(), metav1.ListOptions{})
		if err != nil {
			status = "Not Found"
			color = Red
			passed = false
			checkError(err)
		} else {
			found := false
			for _, pod := range pods.Items {
				if strings.Contains(pod.Name, podTest.Name) {
					found = true
					break
				}
			}

			if found {
				status = "Found"
				color = Green
			} else {
				status = "Not Found"
				color = Red
				passed = false
			}
		}
		fmt.Printf("%s/*%s* - %s%s %s\n", podTest.Namespace, podTest.Name, color, status, Reset)
	}
	return passed
}

File: cluster-check-py/pods.py
from typing import List
import utils

from myTypes import Config, NamespacedEntry
from kubernetes.client import CoreV1Api
from kubernetes.client.rest import ApiException


def validate_pods(config_check: Config, clientset, sce_check: bool, azure_check: bool, oms_check: bool):
    print("---------------------------------------------------------")
    print("Checking for pods")
    print("---------------------------------------------------------")
    passed = True
    if oms_check:
        if not check_pods(config_check.OMS.pods, clientset):
            passed = False
    if sce_check:
        if not check_pods(config_check.SCE.pods, clientset):
            passed = False
    if azure_check:
        if not check_pods(config_check.Azure.pods, clientset):
            passed = False
    print()
    return passed


def check_pods(pods: List[NamespacedEntry], clientset):
    passed = True
    for pod_test in pods:
        status = ""
        color = ""
        try:
            pod_list = clientset.list_namespaced_pod(pod_test.namespace)
            found = False
            for pod in pod_list.items:
                if pod_test.name in pod.metadata.name:
                    found = True
                    break
            if found:
                status = "Found"
                color = utils.Color.GREEN.value
            else:
                status = "Not Found"
                color = utils.Color.RED.value
                passed = False
            print(f"{pod_test.namespace}/*{pod_test.name}* - {color}{status}\033[0m")
        except ApiException as e:
            status = "Not Found"
            color = utils.Color.RED
            passed = False
            utils.check_error(e)
            print(f"{pod_test.namespace}/*{pod_test.name}* - {color}{status}\033[0m")
    return passed